src/plugins/vis_builder/public/visualizations/vega/components/encoding.ts

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { mapFieldTypeToVegaType } from '../utils/helpers';
+import { AxisFormat, AxisFormats } from '../utils/types';
+
+interface BaseEncodingChannel {
+  field?: string;
+  type?: string;
+  legend?: { title: string | null } | null;
+}
+
+export interface AxisEncodingChannel extends BaseEncodingChannel {
+  axis?: { title: string };
+}
+
+export type ColorEncodingChannel = BaseEncodingChannel;
+
+export interface TooltipEncodingChannel extends BaseEncodingChannel {
+  title?: string;
+}
+export interface VegaEncoding {
+  x?: AxisEncodingChannel;
+  y?: AxisEncodingChannel;
+  color?: ColorEncodingChannel;
+  tooltip?: TooltipEncodingChannel | TooltipEncodingChannel[];
+  [key: string]:
+    | BaseEncodingChannel
+    | BaseEncodingChannel[]
+    | TooltipEncodingChannel
+    | TooltipEncodingChannel[]
+    | undefined;
+}
+
+export interface VegaScale {
+  name: string;
+  type: string;
+  domain: {
+    data: string;
+    field: string;
+    filter?: string;
+  };
+  range?: string;
+  padding?: number;
+  nice?: boolean;
+  zero?: boolean;
+}
+
+/**
+ * Builds encoding configuration for Vega-Lite specifications.
+ *
+ * @param {any} dimensions - The dimensions of the data.
+ * @param {AxisFormats} formats - The formatting information for axes.
+ * @returns {VegaEncoding} The Vega-Lite encoding configuration.
+ */
+export const buildVegaLiteEncoding = (dimensions: any, formats: AxisFormats): VegaEncoding => {
+  const { xAxisFormat, xAxisLabel, yAxisFormat, yAxisLabel } = formats;
+  const encoding: VegaEncoding = {};
+
+  // Handle x-axis
+  encoding.x = buildAxisEncoding('x', dimensions.x, xAxisFormat, xAxisLabel);
+
+  // Handle y-axis
+  encoding.y = buildAxisEncoding('y', dimensions.y, yAxisFormat, yAxisLabel);
+
+  // Handle color encoding for multiple y dimensions or series
+  if (dimensions.y) {
+    encoding.color = buildColorEncoding('series', 'nominal');
+  }
+
+  return encoding;
+};
+
+/**
+ * Builds scale configurations for Vega specifications.
+ *
+ * @param {any} dimensions - The dimensions of the data.
+ * @param {any} formats - The formatting information for axes.
+ * @returns {VegaScale[]} The Vega scale configurations.
+ */
+export const buildVegaScales = (dimensions: any, formats: any): VegaScale[] => {
```

Comment:
This function doesn't seem to be using the parameters **dimensions** and **formats** . If so we can clean them up.